### ISSUE TITLE
composer create-project fails on Twig patch application

### ISSUE THREAD

**Issue by tmbritton** [acquia/blt#1770]
My system information:
* Operating system type: OSX
* Operating system version: 10.12.2
* BLT version: Phing 2.15.2

Output of `blt doctor`:
```
blt > doctor:
Drush could not find the docroot.
(The BLT installation doesn't complete because the error is in the composer create-project command)
```

When I run this command:
```
composer create-project --no-interaction acquia/blt-project fancytom
```

I get the following output:
```
Installing acquia/blt-project (8.0.4)
  - Installing acquia/blt-project (8.0.4): Downloading (100%)
Created project in fancytom
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 63 installs, 0 updates, 0 removals
  - Installing wikimedia/composer-merge-plugin (v1.4.0): Loading from cache
  - Installing composer/installers (v1.3.0): Loading from cache
  - Installing grasmash/drupal-security-warning (1.0.0): Loading from cache
  - Installing symfony/yaml (v2.8.24): Loading from cache
  - Installing psr/log (1.0.2): Loading from cache
  - Installing symfony/debug (v3.0.9): Loading from cache
  - Installing symfony/polyfill-mbstring (v1.4.0): Loading from cache
  - Installing symfony/console (v2.8.24): Loading from cache
  - Installing squizlabs/php_codesniffer (2.9.1): Loading from cache
  - Installing sebastian/version (1.0.6): Loading from cache
  - Installing sebastian/global-state (1.1.1): Loading from cache
  - Installing sebastian/recursion-context (1.0.5): Loading from cache
  - Installing sebastian/exporter (1.2.2): Loading from cache
  - Installing sebastian/environment (1.3.8): Loading from cache
  - Installing sebastian/diff (1.4.3): Loading from cache
  - Installing sebastian/comparator (1.2.4): Loading from cache
  - Installing doctrine/instantiator (1.0.5): Loading from cache
  - Installing phpunit/php-text-template (1.2.1): Loading from cache
  - Installing phpunit/phpunit-mock-objects (2.3.8): Loading from cache
  - Installing phpunit/php-timer (1.0.9): Loading from cache
  - Installing phpunit/php-file-iterator (1.4.2): Loading from cache
  - Installing phpunit/php-token-stream (1.4.11): Loading from cache
  - Installing phpunit/php-code-coverage (2.2.4): Loading from cache
  - Installing webmozart/assert (1.2.0): Loading from cache
  - Installing phpdocumentor/reflection-docblock (2.0.5): Loading from cache
  - Installing phpspec/prophecy (v1.7.0): Loading from cache
  - Installing phpunit/phpunit (4.8.36): Loading from cache
  - Installing phing/phing (2.15.2): Loading from cache
  - Installing oomphinc/composer-installers-extender (v1.1.2): Loading from cache
  - Installing dflydev/dot-access-data (v1.1.0): Loading from cache
  - Installing grasmash/yaml-expander (1.1.0): Loading from cache
  - Installing symfony/filesystem (v2.8.24): Loading from cache
  - Installing grasmash/yaml-cli (1.0.2): Loading from cache
  - Installing symfony/event-dispatcher (v2.8.24): Loading from cache
  - Installing webmozart/path-util (2.3.0): Loading from cache
  - Installing symfony/var-dumper (v2.8.24): Loading from cache
  - Installing symfony/finder (v2.8.24): Loading from cache
  - Installing jakub-onderka/php-console-color (0.1): Loading from cache
  - Installing jakub-onderka/php-console-highlighter (v0.3.2): Loading from cache
  - Installing dnoegel/php-xdg-base-dir (0.1): Loading from cache
  - Installing nikic/php-parser (v3.0.6): Loading from cache
  - Installing psy/psysh (v0.8.9): Loading from cache
  - Installing pear/console_table (v1.3.0): Loading from cache
  - Installing consolidation/output-formatters (3.1.10): Loading from cache
  - Installing consolidation/annotated-command (2.4.8): Loading from cache
  - Installing psr/container (1.0.0): Loading from cache
  - Installing container-interop/container-interop (1.2.0): Loading from cache
  - Installing league/container (2.4.1): Loading from cache
  - Installing symfony/process (v3.3.4): Loading from cache
  - Installing consolidation/log (1.0.3): Loading from cache
  - Installing consolidation/config (1.0.0): Loading from cache
  - Installing consolidation/robo (1.1.0): Loading from cache
  - Installing drush/drush (8.1.12): Loading from cache
  - Installing drupal/coder (8.2.12): Loading from cache
  - Installing doctrine/lexer (v1.0.1): Loading from cache
  - Installing doctrine/inflector (v1.1.0): Loading from cache
  - Installing doctrine/collections (v1.4.0): Loading from cache
  - Installing doctrine/cache (v1.6.1): Loading from cache
  - Installing doctrine/annotations (v1.4.0): Loading from cache
  - Installing doctrine/common (v2.7.2): Loading from cache
  - Installing twig/twig (v1.34.4): Loading from cache
  - Installing asm89/twig-lint (1.0.2): Loading from cache
  - Installing acquia/blt (8.8.6): Loading from cache
sebastian/global-state suggests installing ext-uopz (*)
phpunit/php-code-coverage suggests installing ext-xdebug (>=2.2.1)
phpdocumentor/reflection-docblock suggests installing dflydev/markdown (~1.0)
phpdocumentor/reflection-docblock suggests installing erusev/parsedown (~1.0)
phpunit/phpunit suggests installing phpunit/php-invoker (~1.1)
phing/phing suggests installing phpdocumentor/phpdocumentor (Documentation Generator for PHP)
phing/phing suggests installing sebastian/phpcpd (Copy/Paste Detector (CPD) for PHP code)
phing/phing suggests installing phpmd/phpmd (PHP version of PMD tool)
phing/phing suggests installing pdepend/pdepend (PHP version of JDepend)
phing/phing suggests installing phploc/phploc (A tool for quickly measuring the size of a PHP project)
phing/phing suggests installing pear/versioncontrol_svn (A simple OO-style interface for Subversion, the free/open-source version control system)
phing/phing suggests installing pear/versioncontrol_git (A library that provides OO interface to handle Git repository)
phing/phing suggests installing pear/archive_tar (Tar file management class)
phing/phing suggests installing siad007/versioncontrol_hg (A library for interfacing with Mercurial repositories.)
phing/phing suggests installing tedivm/jshrink (Javascript Minifier built in PHP)
symfony/event-dispatcher suggests installing symfony/dependency-injection ()
symfony/event-dispatcher suggests installing symfony/http-kernel ()
symfony/var-dumper suggests installing ext-symfony_debug ()
psy/psysh suggests installing ext-pcntl (Enabling the PCNTL extension makes PsySH a lot happier :))
psy/psysh suggests installing ext-pdo-sqlite (The doc command requires SQLite to work.)
psy/psysh suggests installing hoa/console (A pure PHP readline implementation. You'll want this if your PHP install doesn't already support readline or libedit.)
pear/console_table suggests installing pear/Console_Color2 (>=0.1.2)
consolidation/robo suggests installing pear/archive_tar (Allows tar archives to be created and extracted in taskPack and taskExtract, respectively.)
consolidation/robo suggests installing henrikbjorn/lurker (For monitoring filesystem changes in taskWatch)
consolidation/robo suggests installing patchwork/jsqueeze (For minifying JS files in taskMinify)
consolidation/robo suggests installing natxet/CssMin (For minifying CSS files in taskMinify)
drush/drush suggests installing ext-pcntl (*)
drush/drush suggests installing drush/config-extra (Provides configuration workflow commands, such as config-merge.)
acquia/blt suggests installing hirak/prestissimo (^0.3)
Writing lock file
Generating autoload files
Loading composer repositories with package information
Updating dependencies (including require-dev)
Nothing to install or update
Generating autoload files
Creating BLT templated files...
 > /Users/tom.britton/Sites/fancytom/vendor/acquia/blt/bin/blt create-project
blt > create-project:
   [delete] Deleting 3 files from /Users/tom.britton/Sites/fancytom
blt > prepare-repo:
blt > blt:rsync-template:
     [echo] Copying files from BLT's template into your project...
The value for key 'project.machine_name' was set to 'fancytom' in /Users/tom.britton/Sites/fancytom/blt/project.yml.
     [echo] Merging default configuration into composer.json.
     [echo] Installing new Composer dependencies provided by BLT. This make take a while.
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 63 installs, 0 updates, 0 removals
  - Installing wikimedia/composer-merge-plugin (v1.4.0): Loading from cache
  - Installing composer/installers (v1.3.0): Loading from cache
  - Installing grasmash/drupal-security-warning (1.0.0): Loading from cache
  - Installing symfony/yaml (v2.8.24): Loading from cache
  - Installing psr/log (1.0.2): Loading from cache
  - Installing symfony/debug (v3.0.9): Loading from cache
  - Installing symfony/polyfill-mbstring (v1.4.0): Loading from cache
  - Installing symfony/console (v2.8.24): Loading from cache
  - Installing squizlabs/php_codesniffer (2.9.1): Loading from cache
  - Installing sebastian/version (1.0.6): Loading from cache
  - Installing sebastian/global-state (1.1.1): Loading from cache
  - Installing sebastian/recursion-context (1.0.5): Loading from cache
  - Installing sebastian/exporter (1.2.2): Loading from cache
  - Installing sebastian/environment (1.3.8): Loading from cache
  - Installing sebastian/diff (1.4.3): Loading from cache
  - Installing sebastian/comparator (1.2.4): Loading from cache
  - Installing doctrine/instantiator (1.0.5): Loading from cache
  - Installing phpunit/php-text-template (1.2.1): Loading from cache
  - Installing phpunit/phpunit-mock-objects (2.3.8): Loading from cache
  - Installing phpunit/php-timer (1.0.9): Loading from cache
  - Installing phpunit/php-file-iterator (1.4.2): Loading from cache
  - Installing phpunit/php-token-stream (1.4.11): Loading from cache
  - Installing phpunit/php-code-coverage (2.2.4): Loading from cache
  - Installing webmozart/assert (1.2.0): Loading from cache
  - Installing phpdocumentor/reflection-docblock (2.0.5): Loading from cache
  - Installing phpspec/prophecy (v1.7.0): Loading from cache
  - Installing phpunit/phpunit (4.8.36): Loading from cache
  - Installing phing/phing (2.15.2): Loading from cache
  - Installing oomphinc/composer-installers-extender (v1.1.2): Loading from cache
  - Installing dflydev/dot-access-data (v1.1.0): Loading from cache
  - Installing grasmash/yaml-expander (1.1.0): Loading from cache
  - Installing symfony/filesystem (v2.8.24): Loading from cache
  - Installing grasmash/yaml-cli (1.0.2): Loading from cache
  - Installing symfony/event-dispatcher (v2.8.24): Loading from cache
  - Installing webmozart/path-util (2.3.0): Loading from cache
  - Installing symfony/var-dumper (v2.8.24): Loading from cache
  - Installing symfony/finder (v2.8.24): Loading from cache
  - Installing jakub-onderka/php-console-color (0.1): Loading from cache
  - Installing jakub-onderka/php-console-highlighter (v0.3.2): Loading from cache
  - Installing dnoegel/php-xdg-base-dir (0.1): Loading from cache
  - Installing nikic/php-parser (v3.0.6): Loading from cache
  - Installing psy/psysh (v0.8.9): Loading from cache
  - Installing pear/console_table (v1.3.0): Loading from cache
  - Installing consolidation/output-formatters (3.1.10): Loading from cache
  - Installing consolidation/annotated-command (2.4.8): Loading from cache
  - Installing psr/container (1.0.0): Loading from cache
  - Installing container-interop/container-interop (1.2.0): Loading from cache
  - Installing league/container (2.4.1): Loading from cache
  - Installing symfony/process (v3.3.4): Loading from cache
  - Installing consolidation/log (1.0.3): Loading from cache
  - Installing consolidation/config (1.0.0): Loading from cache
  - Installing consolidation/robo (1.1.0): Loading from cache
  - Installing drush/drush (8.1.12): Loading from cache
  - Installing drupal/coder (8.2.12): Loading from cache
  - Installing doctrine/lexer (v1.0.1): Loading from cache
  - Installing doctrine/inflector (v1.1.0): Loading from cache
  - Installing doctrine/collections (v1.4.0): Loading from cache
  - Installing doctrine/cache (v1.6.1): Loading from cache
  - Installing doctrine/annotations (v1.4.0): Loading from cache
  - Installing doctrine/common (v2.7.2): Loading from cache
  - Installing twig/twig (v1.34.4): Loading from cache
  - Installing asm89/twig-lint (1.0.2): Loading from cache
  - Installing acquia/blt (8.8.6): Loading from cache
sebastian/global-state suggests installing ext-uopz (*)
phpunit/php-code-coverage suggests installing ext-xdebug (>=2.2.1)
phpdocumentor/reflection-docblock suggests installing dflydev/markdown (~1.0)
phpdocumentor/reflection-docblock suggests installing erusev/parsedown (~1.0)
phpunit/phpunit suggests installing phpunit/php-invoker (~1.1)
phing/phing suggests installing phpdocumentor/phpdocumentor (Documentation Generator for PHP)
phing/phing suggests installing sebastian/phpcpd (Copy/Paste Detector (CPD) for PHP code)
phing/phing suggests installing phpmd/phpmd (PHP version of PMD tool)
phing/phing suggests installing pdepend/pdepend (PHP version of JDepend)
phing/phing suggests installing phploc/phploc (A tool for quickly measuring the size of a PHP project)
phing/phing suggests installing pear/versioncontrol_svn (A simple OO-style interface for Subversion, the free/open-source version control system)
phing/phing suggests installing pear/versioncontrol_git (A library that provides OO interface to handle Git repository)
phing/phing suggests installing pear/archive_tar (Tar file management class)
phing/phing suggests installing siad007/versioncontrol_hg (A library for interfacing with Mercurial repositories.)
phing/phing suggests installing tedivm/jshrink (Javascript Minifier built in PHP)
symfony/event-dispatcher suggests installing symfony/dependency-injection ()
symfony/event-dispatcher suggests installing symfony/http-kernel ()
symfony/var-dumper suggests installing ext-symfony_debug ()
psy/psysh suggests installing ext-pcntl (Enabling the PCNTL extension makes PsySH a lot happier :))
psy/psysh suggests installing ext-pdo-sqlite (The doc command requires SQLite to work.)
psy/psysh suggests installing hoa/console (A pure PHP readline implementation. You'll want this if your PHP install doesn't already support readline or libedit.)
pear/console_table suggests installing pear/Console_Color2 (>=0.1.2)
consolidation/robo suggests installing pear/archive_tar (Allows tar archives to be created and extracted in taskPack and taskExtract, respectively.)
consolidation/robo suggests installing henrikbjorn/lurker (For monitoring filesystem changes in taskWatch)
consolidation/robo suggests installing patchwork/jsqueeze (For minifying JS files in taskMinify)
consolidation/robo suggests installing natxet/CssMin (For minifying CSS files in taskMinify)
drush/drush suggests installing ext-pcntl (*)
drush/drush suggests installing drush/config-extra (Provides configuration workflow commands, such as config-merge.)
acquia/blt suggests installing hirak/prestissimo (^0.3)
Writing lock file
Generating autoload files
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 99 installs, 3 updates, 0 removals
  - Installing drupal-composer/drupal-scaffold (2.3.0): Loading from cache
  - Installing cweagans/composer-patches (1.6.1): Loading from cache
Gathering patches for root package.
Gathering patches for dependencies. This might take a minute.
  - Updating doctrine/annotations (v1.4.0 => v1.2.7): Loading from cache
  - Installing zendframework/zend-stdlib (3.1.0): Loading from cache
  - Installing zendframework/zend-escaper (2.5.2): Loading from cache
  - Installing zendframework/zend-feed (2.8.0): Loading from cache
  - Installing psr/http-message (1.0.1): Loading from cache
  - Installing zendframework/zend-diactoros (1.4.0): Loading from cache
  - Installing symfony/translation (v2.8.24): Loading from cache
  - Installing ircmaxell/password-compat (v1.0.4): Loading from cache
  - Installing symfony/polyfill-php55 (v1.4.0): Loading from cache
  - Installing symfony/polyfill-php54 (v1.4.0): Loading from cache
  - Installing symfony/http-foundation (v2.8.24): Loading from cache
  - Installing symfony/psr-http-message-bridge (v1.0.0): Loading from cache
  - Installing symfony/polyfill-iconv (v1.4.0): Loading from cache
  - Updating symfony/debug (v3.0.9 => v2.8.24): Loading from cache
  - Installing symfony/http-kernel (v2.8.24): Loading from cache
  - Installing symfony/routing (v2.8.24): Loading from cache
  - Installing symfony-cmf/routing (1.4.1): Loading from cache
  - Installing stack/builder (v1.0.4): Loading from cache
  - Installing paragonie/random_compat (v2.0.10): Loading from cache
  - Installing masterminds/html5 (2.2.2): Loading from cache
  - Installing guzzlehttp/psr7 (1.4.2): Loading from cache
  - Installing guzzlehttp/promises (v1.3.1): Loading from cache
  - Installing guzzlehttp/guzzle (6.3.0): Loading from cache
  - Installing egulias/email-validator (1.2.14): Loading from cache
  - Installing easyrdf/easyrdf (0.9.1): Loading from cache
  - Installing composer/semver (1.4.2): Loading from cache
  - Installing asm89/stack-cors (1.1.0): Loading from cache
  - Installing symfony/validator (v2.8.24): Loading from cache
  - Installing symfony/serializer (v2.8.24): Loading from cache
  - Updating symfony/process (v3.3.4 => v2.8.24): Loading from cache
  - Installing symfony/dependency-injection (v2.8.24): Loading from cache
  - Installing symfony/polyfill-apcu (v1.4.0): Loading from cache
  - Installing symfony/class-loader (v2.8.24): Loading from cache
  - Installing drupal/core (8.3.5): Loading from cache
  - Applying patches for drupal/core
    https://www.drupal.org/files/issues/no_reliable_method-2752961-29.patch (Clear Twig caches on deploys)
   Could not apply patch! Skipping. The error was: Cannot apply patch https://www.drupal.org/files/issues/no_reliable_method-2752961-29.patch


  [Exception]
  Cannot apply patch Clear Twig caches on deploys (https://www.drupal.org/files/issues/no_reliable_method-2752961-29.patch)!


install [--prefer-source] [--prefer-dist] [--dry-run] [--dev] [--no-dev] [--no-custom-installers] [--no-autoloader] [--no-scripts] [--no-progress] [--no-suggest] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--ignore-platform-reqs] [--] [<packages>]...

[phingcall] /Users/tom.britton/Sites/fancytom/./vendor/acquia/blt/phing/tasks/blt.xml:55:165: Task exited with code 1

BUILD FAILED/Users/tom.britton/Sites/fancytom/./vendor/acquia/blt/phing/tasks/blt.xml:12:37: Execution of the target buildfile failed. Aborting.
; 6 minutes  38.65 seconds
```

And I expected this to happen:

I expect the project creation to be completed.

**Comment by danielnitschelc** [comment]
Dupe of: #1764. It's fixed in dev.

**Comment by grasmash** [comment]
This has been resolved for `create-project` as of 8.8.7 (just released).
